cfme/tests/networks/test_sdn_crud.py

```diff
@@ -41,3 +43,21 @@ def test_sdn_crud(provider, appliance):
     provider.wait_for_delete()
 
     assert not network_provider.exists
+
+
+@pytest.mark.uncollectif(lambda appliance: appliance.version < '5.10',
+                         reason='Delete button was added in CFME 5.10.')
+@pytest.mark.provider([OpenStackProvider], scope='function', override=True)
+def test_router_crud(provider, appliance):
+    router_name = 'test_router_crud-{}'.format(fauxfactory.gen_alphanumeric().lower())
+    routers = appliance.collections.network_routers
+    router = routers.create(router_name, provider, provider.data.networks.cloud_tenant,
+                            provider.data.networks.network_manager)
+    router.delete()
+    provider.refresh_provider_relationships()
+
+    def router_doesnt_exists():
+        router = (r for r in appliance.collections.network_routers.all() if r.name == router_name)
+        return not any(router)
+
+    wait_for(router_doesnt_exists, fail_condition=False, timeout=120, delay=1)
```

Comment:
**Optional** I think it would be good to have an assert here. 

example:
```router_exists, _ = wait_for(lambda: router.exists, fail_condition=False, timeout=120, delay=1)
   assert router_exists
```

or

``` assert not router.exists```